addons/payment/models/payment_provider.py

```diff
@@ -184,6 +197,9 @@ def _valid_field_parameter(self, field, name):
     show_done_msg = fields.Boolean(compute='_compute_view_configuration_fields')
     show_cancel_msg = fields.Boolean(compute='_compute_view_configuration_fields')
 
+    # Conditionally required fields
+    require_available_currency_ids = fields.Boolean(compute='_compute_required_fields')
```

Comment:
Why not use `attrs="{'required':[('code', '=', 'PROVIDER_NAME'), ('state', '!=', 'disabled')]}"/>` in the view ?